http/obmc_shell.hpp

```diff
@@ -0,0 +1,214 @@
+            boost::asio::buffer(inputBuffer.data(), inputBuffer.size()),
+            [this, self(shared_from_this())](boost::beast::error_code ec,
+                                             std::size_t bytesWritten) {
+                BMCWEB_LOG_DEBUG << "Wrote " << bytesWritten << "bytes";
+                doingWrite = false;
+                inputBuffer.erase(0, bytesWritten);
+                if (ec == boost::asio::error::eof)
+                {
+                    session->close("ssh socket port closed");
+                    return;
+                }
+                if (ec)
+                {
+                    session->close("Error in writing to processSSH port");
+                    BMCWEB_LOG_ERROR << "Error in ssh socket write " << ec;
+                    return;
+                }
+                doWrite();
+            });
+    }
+
+    void doRead()
+    {
+        if (!session)
+        {
+            BMCWEB_LOG_DEBUG << "session is closed";
+            return;
+        }
+        streamFileDescriptor.async_read_some(
+            boost::asio::buffer(outputBuffer.data(), outputBuffer.size()),
+            [this, self(shared_from_this())](
+                const boost::system::error_code& ec, std::size_t bytesRead) {
+                BMCWEB_LOG_DEBUG << "Read done.  Read " << bytesRead
+                                 << " bytes";
+                if (!session)
+                {
+                    BMCWEB_LOG_DEBUG << "session is closed";
+                    return;
+                }
+                if (ec)
+                {
+                    BMCWEB_LOG_ERROR << "Couldn't read from ssh port: " << ec;
+                    session->close("Error in connecting to ssh port");
+                    return;
+                }
+                std::string_view payload(outputBuffer.data(), bytesRead);
+                session->sendBinary(payload);
+                doRead();
+            });
+    }
+
+    crow::websocket::Connection* session;
+    boost::asio::posix::stream_descriptor streamFileDescriptor;
+    bool doingWrite;
+    int ttyFileDescriptor;
+    pid_t pid;
+
+    std::array<char, 4096> outputBuffer;
+    std::string inputBuffer;
+};
+
+static std::map<crow::websocket::Connection*, std::shared_ptr<Handler>>
+    mapHandler;
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(app, "/bmc-console")
+        .privileges({{"ConfigureManager"}})
+        .websocket()
+        .onopen([](crow::websocket::Connection& conn,
+                   const std::shared_ptr<bmcweb::AsyncResp>&) {
+            BMCWEB_LOG_DEBUG << "Connection " << &conn << " opened";
+            if (auto it = mapHandler.find(&conn); it == mapHandler.end())
+            {
+                auto insertData =
+                    mapHandler.emplace(&conn, std::make_shared<Handler>(&conn));
+                if (std::get<bool>(insertData))
+                {
+                    std::get<0>(insertData)->second->connect();
```

Comment:
At this point, you should pass the username of the authenticated user into the connect() method.
This is needed so the new command shell will run as the user who initiated the connection.
You can get the username from `conn->getUserName()`.  Reference: https://github.com/openbmc/bmcweb/blob/master/http/websocket.hpp